pkg/services/hcloud/server/server.go

```diff
@@ -492,24 +534,91 @@ func (s *Service) createServer(ctx context.Context) (*hcloud.Server, error) {
 	}
 
 	// set ssh keys to status
-	s.scope.HCloudMachine.Status.SSHKeys = sshKeySpecs
+	s.scope.HCloudMachine.Status.SSHKeys = caphSSHKeys
 
 	conditions.MarkTrue(s.scope.HCloudMachine, infrav1.ServerCreateSucceededCondition)
 	record.Eventf(s.scope.HCloudMachine, "SuccessfulCreate", "Created new server %s with ID %d", server.Name, server.ID)
 	return server, nil
 }
 
-func (s *Service) getServerImage(ctx context.Context) (*hcloud.Image, error) {
+// getSSHKeys returns the sshKey slice for the status and the sshKey slice for the createServer API
```

Comment:
please improve this explanation further. It takes the ssh keys referenced in the spec of either hcloudMachine or HetznerCluster, as well as the hetzner secret. 

It then lists all available ssh keys that have been uploaded in Hcloud. It ensures that all keys referenced in the spec or the secret exist in HCloud and returns an error if that is not the case. 

It then returns for each ssh key referenced in the spec or the secret two objects, one as input for the hcloud create server flow, another one for adding the ssh key to the status of the HcloudMachine object.